### DISCUSSION TITLE
Change image creation default from png to webp.

### DISCUSSION THREAD

**Discussion by ejboettcher** [quarto-dev/quarto-cli#2714] [Q&A]
First thank you for Quarto.  I find it refreshing and easier to use than Jekyll or 11ty.

However, 11ty with Tailwind will generate new images for static sites that are optimized for both mobile and computer using web friendly image format .wepb.  This helps makes the site feel more responsive.

Is there any way you can change the default image that is created from code say from a python plot to webp.  Quarto currently saves them as a LARGE png file?  Even switching to jpg can save bandwidth.  Bonus, if you can generate two static images: one for mobile and another desktop.

Thanks again for putting quarto together.

**Comment by cscheid** [comment]
We already have multiformat figures, so if I understand your proposal correctly, this should be doable as a quarto filter extension that would look at the generated HTML and create the new images as needed (this will be significantly easier in 1.3 as well). @dragonstyle how much figure work do we do in our html TS postprocessor? I'm wondering if that could be a snag.

I'm somewhat reluctant to move all of our images to format controlled by Google (and .jpg is lossy and problematic), but a filter extension seems ideal for this.

**Reply by ejboettcher** [reply]
There is no perfect image format.  But, webp is being pushed hard for websites.  

I do like the idea for a user defined image output format option for images that are generated via quarto.  There are many cases where you really want the high resolution.  

That said, I just finished teaching a python class and I used quarto to make the slides and the companion website (again thank you).  In all but a few of the images,  I could have gotten away with a jpg.  

Images I added to my markdown files I used webp, but most of the images in my slides came from python code rendering something.  The quarto -> jupyter -> RevealJS or static site produced .png images.  This is where I would like to option to have the export be webp or jpg.

**Reply by dragonstyle** [reply]
I think the post-processors should be ok with a filter swapping those out - we do resource discovery, perhaps wrap some figures or adorn them with classes, and discover images in rendered output for things like social metadata or listings. But I think in general things should should work fine with your proposed approach!

**Reply by klausbrunner** [reply]
> Images I added to my markdown files I used webp, but most of the images in my slides came from python code rendering something. The quarto -> jupyter -> RevealJS or static site produced .png images. This is where I would like to option to have the export be webp or jpg.

I'm in a similar situation and built a little filter extension for this purpose, as suggested: https://github.com/klausbrunner/convert-to-webp 

Works well for me, but I've only tested on a small blog/website project so far. Obviously, having the code chunks generate webp directly would be the best solution, but this is a (mostly) non-intrusive workaround for the time being.

**Reply by mcanouil** [reply]
Similarly, there is PDF to WebP Quarto extension:

- https://github.com/roaldarbol/pdf-to-image

Source: https://m.canouil.dev/quarto-extensions/
